azurerm/resource_arm_app_service.go

```diff
@@ -0,0 +1,165 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/Azure/azure-sdk-for-go/arm/web"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceArmAppService() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmAppServiceCreateUpdate,
+		Read:   resourceArmAppServiceRead,
+		Update: resourceArmAppServiceCreateUpdate,
+		Delete: resourceArmAppServiceDelete,
+
+		Schema: map[string]*schema.Schema{
+			"resource_group_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"skip_dns_registration": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"skip_custom_domain_verification": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"force_dns_registration": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"ttl_in_seconds": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "",
+			},
+			"app_service_plan_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"always_on": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"location": locationSchema(),
+			"tags":     tagsSchema(),
+		},
+	}
+}
+
+func resourceArmAppServiceCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient)
+	appClient := client.appsClient
+
+	log.Printf("[INFO] preparing arguments for Azure ARM Web App creation.")
+
+	resGroup := d.Get("resource_group_name").(string)
+	name := d.Get("name").(string)
+	location := d.Get("location").(string)
+	skipDNSRegistration := d.Get("skip_dns_registration").(bool)
+	skipCustomDomainVerification := d.Get("skip_custom_domain_verification").(bool)
+	forceDNSRegistration := d.Get("force_dns_registration").(bool)
+	ttlInSeconds := d.Get("ttl_in_seconds").(string)
+	tags := d.Get("tags").(map[string]interface{})
+
+	siteConfig := web.SiteConfig{}
```

Comment:
[Looking at the available fields](https://github.com/Azure/azure-sdk-for-go/blob/master/arm/web/models.go#L2882), it feels like it might be worth splitting this out into a separate block?